tests/blacken_docs_test.py

```diff
@@ -725,3 +725,243 @@ def test_format_src_rst_pycon_comment_before_promopt():
+    assert after == (
+        '```pycon\n'
+        '>>> if True:\n'
+        '...     pass\n'
+        '...\n'
+        '```\n'
+    )
+
+
+def test_format_src_markdown_pycon_code_block_is_final_line2():
+    before = (
+        '```pycon\n'
+        '>>> if True:\n'
+        '...   pass\n'
+        '```\n'
+    )
+    after, _ = blacken_docs.format_str(before, BLACK_MODE)
+    assert after == (
+        '```pycon\n'
+        '>>> if True:\n'
+        '...     pass\n'
+        '...\n'
+        '```\n'
+    )
+
+
+def test_format_src_markdown_pycon_nested_def1():
+    before = (
+        '```pycon\n'
+        '>>> if True:\n'
+        '...     def f(): pass\n'
+        '...\n'
+        '```\n'
+    )
+    after, _ = blacken_docs.format_str(before, BLACK_MODE)
+    assert after == (
+        '```pycon\n'
+        '>>> if True:\n'
+        '...     def f():\n'
+        '...         pass\n'
+        '...\n'
+        '```\n'
+    )
+
+
+def test_format_src_markdown_pycon_nested_def2():
+    before = (
+        '```pycon\n'
+        '>>> if True:\n'
+        '...     def f(): pass\n'
+        '```\n'
+    )
+    after, _ = blacken_docs.format_str(before, BLACK_MODE)
+    assert after == (
+        '```pycon\n'
+        '>>> if True:\n'
+        '...     def f():\n'
+        '...         pass\n'
+        '...\n'
+        '```\n'
+    )
+
+
+def test_format_src_markdown_pycon_empty_line():
+    before = (
+        '```pycon\n'
+        '>>> l = [\n'
+        '...\n'
+        '...     1,\n'
+        '... ]\n'
+        '```\n'
+    )
+    after, _ = blacken_docs.format_str(before, BLACK_MODE)
+    assert after == (
+        '```pycon\n'
+        '>>> l = [\n'
+        '...     1,\n'
+        '... ]\n'
+        '```\n'
+    )
+
+
+def test_format_src_markdown_pycon_preserves_output_indentation():
+    before = (
+        '```pycon\n'
+        '>>> 1 / 0\n'
+        'Traceback (most recent call last):\n'
+        '  File "<stdin>", line 1, in <module>\n'
+        'ZeroDivisionError: division by zero\n'
+        '```\n'
+    )
+    after, _ = blacken_docs.format_str(before, BLACK_MODE)
+    assert after == before
+
+
+def test_format_src_markdown_pycon_elided_traceback():
+    before = (
+        '```pycon\n'
+        '>>> 1 / 0\n'
+        'Traceback (most recent call last):\n'
+        '  ...\n'
+        'ZeroDivisionError: division by zero\n'
+        '```\n'
+    )
+    after, _ = blacken_docs.format_str(before, BLACK_MODE)
+    assert after == before
+
+
+def test_format_src_markdown_pycon_no_prompt():
+    before = (
+        '```pycon\n'
+        'pass\n'
+        '```\n'
+    )
+    after, _ = blacken_docs.format_str(before, BLACK_MODE)
+    assert after == before
+
+
+def test_format_src_markdown_pycon_comment_before_promopt():
+    before = (
+        '```pycon\n'
+        '# Comment about next line\n'
+        '>>> pass\n'
+        '```\n'
+    )
+    after, _ = blacken_docs.format_str(before, BLACK_MODE)
+    assert after == before
```

Comment:
you don't need all these tests, please just one test demonstrating your change

Reply:
I've kept just the first 1